docs/source/model_design/programming_vivarium/artifact_building/index.rst

```diff
@@ -29,12 +29,156 @@ Artifact Building
 
 .. contents::
    :local:
-   :depth: 1
+   :depth: 2
 
-.. todo::
+A data artifact is a bundle of input data associated with a particular
+model. It is typically stored as an :code:`hdf` file on disk with very
+particular formatting. This file is then used by the vivarium simulations
+to fill in all the relevant parameter data.
 
-  Add to this page. Include how-to type information on building the artifact, targeted to a research team audience. Include links to the engineering documentation as well. 
+There are also artifact building instructions in the `vivarium tutorial docs <https://vivarium.readthedocs.io/en/latest/tutorials/artifact.html>`_. However,
+these are largely written for a software developer, so here we endeavor to
+create documentation from a research team perspective.
 
-  Review and integrate this `engineering page <https://vivarium.readthedocs.io/en/latest/tutorials/artifact.html>`_. 
+Understanding Data Keys
+-----------------------
 
+A data key in a simulation is a key word that defines a set of
+data to be saved in the artifact. For example, :code:`population.location`,
+:code:`cause.measles.prevalence`, or :code:`risk_factor.child_wasting.exposure`.
 
+You should be used to seeing data keys like this in the artifact, since
+research team members often use these in V&V. In fact, this is how the data
+is called to create the artifact, and called from the artifact by research to
+test the model.
+
+These data keys show up in two places:
+
+#. The :code:`data_keys.py` file
+#. The :code:`loader.py` file
+
+In :code:`data_keys.py`, all data keys are defined. At the bottom in
+the :code:`MAKE_ARTIFACT_KEY_GROUPS` you'll see the data key groups -
+things like :code:`POPULATION`, :code:`MEASLES`, or :code:`WASTING`
+from the examples above. The code above in the rest of the documentation
+is designed to take inputs from :code:`loader.py` and return the correct
+data key string.
+
+At the top of :code:`loader.py` you will see :code:`mapping`. The map provides
+inputs for :code:`data_keys.py` which returns the data key name. The second
+part of the line in :code:`mapping` is the function where the data key
```

Comment:
I wonder if we should reference the template repo here for context on where this code comes from? It may be unclear how to get started with these files when starting a new project versus working in an existing one

Reply:
Oh, I see that you do reference this later on the page, but I think it would be helpful to include that up higher maybe in the intro for context before getting to this section.

Reply:
Good idea - I added it to the intro actually so its right at the top as a resource.